### ISSUE TITLE
OverFitting to rFID ImageNet?

### ISSUE THREAD

**Issue by jiasenlu** [bytetriper/RAE#13]
Hi, 

Very interesting paper, but it seems the PSNR is very low. (PSNR: 18.9438, rFID: 0.5776)

https://github.com/bytetriper/RAE/issues/11#issuecomment-3405022602

I also tried to reconstruct some human faces and text, and it seems it is not working on these images at all. 

original 

<img width="300" height="200" alt="Image" src="https://github.com/user-attachments/assets/5900a133-915b-40ca-9a1a-25286879e1c8" />

RAE-Dino

<img width="256" height="256" alt="Image" src="https://github.com/user-attachments/assets/5cb56716-1806-469b-b7b9-ce37c3c5df17" />

RAE-Siglip

<img width="256" height="256" alt="Image" src="https://github.com/user-attachments/assets/9e0001a1-6a43-4df4-be81-5864c4766d8c" />

Did I miss anything? Do you think RAE is overfiting the rFID metric on ImageNet? Maybe finetune the Encoder can improve it? 

Thanks.

**Comment by tsb0601** [comment]
Hi, 

Regarding faces and text reconstruction:
- You're right to notice this limitation! Since we only trained the decoder on ImageNet, it naturally mimics ImageNet's distribution, which contains relatively few human faces and even fewer images with dense text. So the poor reconstruction quality on these types of images is actually expected behavior given our training data. The decoder hasn't learned to represent these domains well. SD-VAE, for example, is trained on dataset beyond imagenet. We believe it is definitely a TODO item for the future. 

Regarding PSNR/SSIM:
- Our training objective for RAE doesn't focus on achieving high PSNR or SSIM. In fact, several design choices intentionally trade off reconstruction fidelity for better generative performance. For example, in Sec. 4.3 we train the decoder with noisy latents, which naturally reduces low-level reconstruction quality but enhances image generation. We also resize DINOv2 inputs to 224×224 since only the patch-14 variant is available—further lowering PSNR/SSIM.

**Comment by jiasenlu** [comment]
> Hi,
> 
> Regarding faces and text reconstruction:
> 
> * You're right to notice this limitation! Since we only trained the decoder on ImageNet, it naturally mimics ImageNet's distribution, which contains relatively few human faces and even fewer images with dense text. So the poor reconstruction quality on these types of images is actually expected behavior given our training data. The decoder hasn't learned to represent these domains well. SD-VAE, for example, is trained on dataset beyond imagenet. We believe it is definitely a TODO item for the future.
> 

Thanks, do you think the limiation is caused by training data? or the encoder is frozen and does not contain the detail information? 

> Regarding PSNR/SSIM:
> 
> * Our training objective for RAE doesn't focus on achieving high PSNR or SSIM. In fact, several design choices intentionally trade off reconstruction fidelity for better generative performance. For example, in Sec. 4.3 we train the decoder with noisy latents, which naturally reduces low-level reconstruction quality but enhances image generation. We also resize DINOv2 inputs to 224×224 since only the patch-14 variant is available—further lowering PSNR/SSIM.


That make sense, however, this does feel like the training target to improve rFID, gFID, I think face and text renderding is also very important for image generation tasks, not only imagenet.

**Comment by bytetriper** [comment]
Hi @jiasenlu and @Tristan-mc-q,

I’d also like to note that RAE should not be directly compared with SD-VAE (or FLUX-VAE), especially when it comes to reconstruction outside the ImageNet distribution. Poor reconstruction of small faces or text is a common limitation of autoencoders trained solely on ImageNet—simply because such examples are rare or even absent in the dataset. Our MAE-based decoder performs much better, but it’s still constrained by the training data.

In contrast, SD-VAE and FLUX-VAE are trained on much more diverse datasets. For instance, SD-VAE is trained on around 17M images, with a significant portion containing human faces. If your use case involves such content, it would definitely be preferable to train RAE on a broader dataset rather than ImageNet.

**Comment by jiasenlu** [comment]
Thanks for the note @bytetriper @tsb0601. I agree that imagenet data maybe the bottleneck. Here is the result of VA-VAE tokenizer, and they aslo trained only using imagenet data. 

Source
<img width="256" height="256" alt="Image" src="https://github.com/user-attachments/assets/a1e906e1-11e6-4f54-8adf-4f777edcc44d" />

VAVAE-reconstruct
<img width="256" height="256" alt="Image" src="https://github.com/user-attachments/assets/7c29089a-3f80-4ac1-9175-6fcdb3f5d327" />

It also has some artifact, but it seems much better compare to RAE. 

RAE-dino
<img width="256" height="256" alt="Image" src="https://github.com/user-attachments/assets/e9714a7a-d3f5-4d54-a4d0-e0febdb51530" />


I think this is a great idea, and promising direction to unify the generation and understanding features. It will be great addition, if you can train on larger dataset and show that will work on those cases.

**Comment by qsh-zh** [comment]
hi @jiasenlu , is your psnr number based on imagenet training dataset or  ood dataset?
